generate_metap_agg.py

```diff
@@ -1,193 +1,465 @@
+        self, collection: str, filter="", max_page_size=100, fields=""
+    ):
+        """General function to get results from the API using the collection endpoint with optional filter and fields
+        
+        Parameters
+        ----------
+        collection : str
+            Collection name to query, e.g. "functional_annotation_agg"
+        filter : str, optional
+            Filter to apply to the query written in JSON format for MongoDB
+            e.g. '{"metagenome_annotation_id":{"$regex":"wfmp"}}'
+            Default is an empty string, which does not apply a filter
+        max_page_size : int, optional
+            Maximum number of records to return in a single page
+            Default is 100
+        fields : str, optional
+            Fields to return in the query, separated by commas without spaces if multiple
+            e.g. "id,data_object_type,url"
+            Default is an empty string, which returns all fields
+        """
+
+        # Get initial results (before next_page_token is given in the results)
+        result_list = []
+        og_url = f"{self.base_url}/nmdcschema/{collection}?&filter={filter}&max_page_size={max_page_size}&projection={fields}"
+        resp = requests.get(og_url)
+        initial_data = resp.json()
+        results = initial_data.get("resources", [])
+
+        if results == []:
+            # if no results are returned
+            return result_list
+
+        # append first page of results to an empty list
+        for result in results:
+            result_list.append(result)
+
+        # if there are multiple pages of results returned
+        if initial_data.get("next_page_token"):
+            next_page_token = initial_data["next_page_token"]
+
+            while True:
+                url = f"{self.base_url}/nmdcschema/{collection}?&filter={filter}&max_page_size={max_page_size}&next_page_token={next_page_token}&projection={fields}"
+                response = requests.get(url)
+                data_next = response.json()
+
+                results = data_next.get("resources", [])
+                result_list.extend(results)
+                next_page_token = data_next.get("next_page_token")
+
+                if not next_page_token:
+                    break
+
+        return result_list
+    
+    def get_previously_aggregated_workflow_ids(self):
+        """Function to return all ids of workflow execution ids that have already been aggregated.
+
+        Uses the aggregation_filter attribute to filter the results for subclasses.
+
+        Returns
+        -------
+        list
+            List of workflow ids that have already been aggregated
+        """
+        agg_col = self.get_results(
+            collection="functional_annotation_agg",
+            filter=self.aggregation_filter,
+            max_page_size=1000,
+            fields="metagenome_annotation_id",
+        )
+        ids = list(set([x["metagenome_annotation_id"] for x in agg_col]))
```

Comment:
shouldn't this be was_generated_by based on the latest schema changes?

Reply:
Yes, I'll change throughout once `was_generated_by` is changed in the production database.  Until then, I've been testing with `metagenome_annotation_id`.

Reply:
nm, i see you todo comment on line 8

Reply:
Updated now.